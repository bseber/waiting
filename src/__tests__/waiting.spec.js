import React from "react";
import { renderShallow, renderIntoDocument } from "../../test/testHelpers";
import Waiting from "../waiting";

jest.useFakeTimers();

describe("Waiting", () => {
    it("renders default", () => {
        const root = document.createElement("div");
        renderIntoDocument(<Waiting />, root);
        expect(root.firstChild).toMatchSnapshot();
    });

    it("renders null for loading=false", () => {
        const tree = renderShallow(<Waiting loading={false} />);
        expect(tree).toBeNull();
    });

    it("does not render loading info when flag is toggled again within a short delay", () => {
        const root = document.createElement("div");

        renderIntoDocument(<Waiting loading={false} />, root);
        expect(root.textContent).toBe("");

        renderIntoDocument(<Waiting loading={true} />, root);
        jest.runTimersToTime(99);
        expect(root.textContent).toBe("");

        renderIntoDocument(<Waiting loading={false} />, root);
        jest.runOnlyPendingTimers();
        expect(root.textContent).toBe("");
    });

    it("does not delay rendering when loading flag is toggled to 'false'", () => {
        const root = document.createElement("div");

        renderIntoDocument(<Waiting loading={true} />, root);
        expect(root.textContent).toBe("loading...");

        renderIntoDocument(<Waiting loading={false} />, root);
        // no "jest.runTimersToTime(99);" on purpose
        // since loading info should be removed at once
        expect(root.textContent).toBe("");
    });

    it("invokes render property to render content", () => {
        const root = document.createElement("div");
        renderIntoDocument(
            <Waiting
                loading={false}
                render={() => (
                    <ul>
                        <li>bruce</li>
                        <li>clark</li>
                    </ul>
                )}
            />,
            root,
        );
        expect(root.firstChild).toMatchSnapshot();
    });
});
