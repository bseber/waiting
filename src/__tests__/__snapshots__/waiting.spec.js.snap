// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Waiting invokes render property to render content 1`] = `
<ul>
  <li>
    bruce
  </li>
  <li>
    clark
  </li>
</ul>
`;

exports[`Waiting renders default 1`] = `
<div>
  loading...
</div>
`;
